Cabal/doc/developing-packages.rst

```diff
@@ -1,83 +1,119 @@
+.. highlight:: console
 
 ::
 
-    $ cabal init
-    Package name [default "proglet"]?
-    Package version [default "0.1"]?
-    ...
+    $ cabal init --non-interactive
+    # You can also use -n which is the short version of --non-interactive
 
-It also asks questions about various other bits of package metadata. For
-a package that you never intend to distribute to others, these fields
-can be left blank.
+If you want, you can also try out the interactive mode, for now chose
+"Executable" when asked what type of package you want to build.
 
-One of the important questions is whether the package contains a library
-or an executable. Libraries are collections of Haskell modules that can
-be re-used by other Haskell libraries and programs, while executables
-are standalone programs.
+.. highlight:: console
 
 ::
 
+    $ cabal init --interactive
+    ...
     What does the package build:
-       1) Library
-       2) Executable
+      1) Executable
+      2) Library
+      3) Library and Executable
     Your choice?
 
-For the moment these are the only choices. For more complex packages
-(e.g. a library and multiple executables or test suites) the ``.cabal``
-file can be edited afterwards.
 
-Finally, ``cabal init`` creates the initial ``proglet.cabal`` and
-``Setup.hs`` files, and depending on your choice of license, a
-``LICENSE`` file as well.
+This will create the ``Main.hs`` Haskell file and a few others that are
+part of all Cabal packages. By default ``cabal init`` will create an
+executable in the top level project directory.
 
 ::
 
-    Generating LICENSE...
     Generating Setup.hs...
+    Generating CHANGELOG.md...
+    Generating Main.hs...
     Generating proglet.cabal...
 
-    You may want to edit the .cabal file and add a Description field.
+Check out the files created using the ``ls`` command.
+
+::
+
+   $ ls
+   CHANGELOG.md  Main.hs  proglet.cabal  Setup.hs
+
+
+Running the program
+-------------------
+
+Now that we have our Haskell code and the extra files that Cabal needs we
+can build and run our project.
+
+::
+
+   $ cabal build
+   Resolving dependencies...
+   ...
+   Linking /path/to/proglet ...
+
+   $ cabal run
+   <Stuff about resolving dependencies, build profile, compliling, linking>
+   Hello, Haskell!
+
+Since we have an executable we can use ``cabal run`` which will build our
+executable (and re-build it if we've made any changes) and then run the
+binary.
+
+
+About the Cabal package structure
+---------------------------------
+
+It is assumed that all the files that make up a package live under a common
+project root directory (apart from external dependencies). This simple
+example will have all the project files in one directory, but most packages
+will use one or more subdirectories.
+
+Cabal needs two extra files in the project's root directory:
+
+-  ``proglet.cabal``: contains package metadata and build information.
+
+-  ``Setup.hs``: contains a few standardized lines of code, but can be
```

Comment:
In fact, when it's `build-type: Simple`, the contents **have to be standard**. Better would be for `cabal init` not to create `Setup.hs` at all.

Reply:
Acknowledged. Currently `cabal init` still creates the `Setup.hs` file, if you can create an issue and assign it to me I can clean that up as a follow-up to this PR and update the docs.